packit_service/worker/handlers/open_scan_hub.py

```diff
@@ -0,0 +1,170 @@
+        super().__init__(**kwargs)
+        self.event: Union[
+            OpenScanHubTaskFinishedEvent | OpenScanHubTaskStartedEvent
+        ] = self.data.to_event()
+
+    @staticmethod
+    def get_checkers() -> Tuple[Type[Checker], ...]:
+        return (RawhideX86Target,)
+
+    def get_helper(self) -> OpenScanHubHelper:
+        build_helper = CoprBuildJobHelper(
+            service_config=self.service_config,
+            package_config=self.package_config,
+            project=self.project,
+            metadata=self.data,
+            db_project_event=self.data.db_project_event,
+            job_config=self.job_config,
+            celery_task=self.celery_task,
+        )
+
+        scan_helper = OpenScanHubHelper(
+            copr_build_helper=build_helper, build=self.event.build
+        )
+        return scan_helper
+
+    def check_scan_and_build(self):
+        task_id = self.data.event_dict["task_id"]
+        if not self.event.scan or not self.event.build:
+            return TaskResults(
+                success=True,
+                details={
+                    "msg": f"Scan {task_id} not found or not associated with a Copr build"
+                },
+            )
+        elif not self.job_config:
+            return TaskResults(
+                success=True,
+                details={
+                    "msg": (
+                        "No job configuration found for OpenScanHub task"
+                        f" in {self.project.repo}"
+                    )
+                },
+            )
+
+
+@configured_as(job_type=JobType.copr_build)
+@reacts_to(OpenScanHubTaskFinishedEvent)
+class OpenScanHubTaskFinishedHandler(
+    OpenScanHubAbstractHandler,
+):
+    event: OpenScanHubTaskFinishedEvent
+    task_name = TaskName.openscanhub_task_finished
+
+    def run(self) -> TaskResults:
+        self.check_scan_and_build()
+
+        if self.event.status == OpenScanHubTaskFinishedEvent.Status.success:
+            state = BaseCommitStatus.success
+            description = (
+                "Scan in OpenScanHub is finished. Check the URL for more details."
+            )
+            external_links = {
+                "Added issues": self.event.issues_added_url,
+                "Fixed issues": self.event.issues_fixed_url,
+                "Scan results": self.event.scan_results_url,
+            }
+            self.event.scan.set_status(OSHScanStatus.succeeded)
+            self.event.scan.set_issues_added_url(self.event.issues_added_url)
+            self.event.scan.set_issues_fixed_url(self.event.issues_fixed_url)
+            self.event.scan.set_scan_results_url(self.event.scan_results_url)
+        else:
+            state = BaseCommitStatus.neutral
+            description = (
+                f"Scan in OpenScanHub is finished in a {self.event.status} state."
+            )
+            external_links = {}
+            if self.event.status == OpenScanHubTaskFinishedEvent.Status.cancel:
```

Comment:
what about the `interrupt` state here?

Reply:
I thought that from a Packit point of view the `interrupted` and `canceled` states can have the same "view". I mapped them both to a `canceled` state in our db. 
But the details are available in the Open Scan Hub logs and also the message we give back has the detail if this was canceled or interrupted...
But if we want to map all the states in Packit too, I can add the missing one.

Reply:
ok, that makes sense, thanks for the explanation!